### PR TITLE
[IOCOM-253,IOCOM-258] Lollipop middleware and third party message endpoints with lollipop

### PR DESCRIPTION
**Pull request by Vangaorth**
## Short description
This PR refactors lollipop checks and adds the new lollipop middleware.

## List of changes proposed in this pull request
- Lollipop logic in `src/routers/features/lollipop/index.ts` has been moved to `src/middleware/lollipopMiddleware.ts`
- The following endpoints use the lollipop middleware:
  - message precondition
  - third party message
  - third party message attachments
  - lollipop playground
- global configuration to disable every lollipop middleware in `/src/config.ts`
- README.md updated with instructions on how to use lollipop

## How to test
- With the global lollipop configuration enabled, check that the previous endpoints do not work with improper headers.
- Repeat the same test but with global lollipop configuration disabled (endpoints shall work)
- Using the latest master deploy of io-app (with lollipop enabled globally), check that, for any PN message:
  - preconditions are retrieved
  - third party message is retrieved
  - attachments are retrieved

### CODE REVIEW

File: src/middleware/lollipopMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          pipe(
+            await verifyLollipopSignatureHeader(request, response),
+            E.foldW(
+              error => response.status(error.code).send(error.problemJson),
+              _ => nextMiddleware(request, response)
+            )
+          )
+      )
+    );
+  };
```
Comment (shadowsheep1) [inline src/middleware/lollipopMiddleware.ts:44]: Here we could avoid `async/await` with a more `fp-ts` idiomization... 

```suggestion
export const lollipopMiddleware =
  (nextMiddleware: (embeddedRequest: Request, _: Response) => void) =>
  (request: Request, response: Response) => {
    pipe(
      isLollipopConfigEnabled(),
      B.fold(
        () => nextMiddleware(request, response),
        () =>
          pipe(
            TE.tryCatch(
              () => verifyLollipopSignatureHeader(request, response),
              _ => _ as Error
            ),
            TE.map(verificationResult =>
              pipe(
                verificationResult,
                E.foldW(
                  error => response.status(error.code).send(error.problemJson),
                  _ => nextMiddleware(request, response)
                )
              )
            )
          )()
      )
    );
  };
```
Is a bit more verbose, as usual... 😊. What do you think?

Comment (Vangaorth) [reply src/middleware/lollipopMiddleware.ts:44]: Yeah I prefer the one you are suggesting, let's adopt it

File: src/middleware/lollipopMiddleware.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                )
+              )
+            )
+          )()
+      )
+    );
+  };
+
+const verifyLollipopSignatureHeader = async (req: Request, _: Response) =>
+  await pipe(
```
Comment (shadowsheep1) [inline src/middleware/lollipopMiddleware.ts:55]: We can now remove `async/await` here too, without any further modification. Am I right?

Comment (Vangaorth) [reply src/middleware/lollipopMiddleware.ts]: Totally right, corrected

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**